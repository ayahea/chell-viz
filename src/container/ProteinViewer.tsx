import * as fs from 'fs';
import * as NGL from 'ngl';
import * as React from 'react';

import { NGLComponent } from '../component/NGLComponent';
import { SequenceViewer } from '../component/SequenceViewer';

export interface IProteinViewerState {
  data?: NGL.Structure;
  resname?: string;
}

export class ProteinViewer extends React.Component<any, IProteinViewerState> {
  constructor(props: any) {
    super(props);
    this.state = {
      data: undefined,
    };
  }

  public async componentDidMount() {
    const fileLoc = 'assets/1fqg.pdb';
    const data = (await NGL.autoLoad(fileLoc)) as NGL.Structure;
    this.setState({
      data,
    });
  }

  public render() {
    const onHoverPickCallback = (resname: string) => {
      this.setState({
        resname,
      });
    };

    return (
      <div id="ProteinViewer">
        <NGLComponent data={this.state.data} onHoverPickCallback={onHoverPickCallback} />
        <SequenceViewer data={this.state.data} selectedResname={this.state.resname} />
      </div>
    );
  }
}
