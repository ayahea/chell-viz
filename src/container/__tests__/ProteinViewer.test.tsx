import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { ProteinViewer } from '../ProteinViewer';

describe('ProteinViewer', () => {
  test('Should match existing snapshot when given no data.', () => {
    expect(toJson(shallow(<ProteinViewer />))).toMatchSnapshot();
  });
});
