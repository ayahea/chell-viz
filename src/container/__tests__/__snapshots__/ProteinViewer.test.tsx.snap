// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProteinViewer Should match existing snapshot when given no data. 1`] = `
<div
  id="ProteinViewer"
>
  <Component />
  <SequenceViewer />
</div>
`;
